runtime/interpreter/interpreter_statement.go

```diff
@@ -221,14 +227,14 @@ func (interpreter *Interpreter) VisitSwitchStatement(switchStatement *ast.Switch
 
 		result := interpreter.evalExpression(switchCase.Expression)
 
-		caseValue := result.(EquatableValue)
+		caseValue, ok := result.(EquatableValue)
 
 		// If the test value and case values are equal,
 		// evaluate the case's statements
 
 		getLocationRange := locationRangeGetter(interpreter.Location, switchCase.Expression)
 
-		if testValue.Equal(interpreter, getLocationRange, caseValue) {
+		if ok && testValue.Equal(interpreter, getLocationRange, caseValue) {
```

Comment:
maybe continue `if !ok`, so that adding new statements below will have minimum side effects.